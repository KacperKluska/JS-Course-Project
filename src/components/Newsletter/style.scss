@import "../../colors.scss";

.newsletter {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.newsletterForm {
  display: flex;
  flex-direction: row;

  margin-top: 0.5rem;
}

.newsletterInput {
  height: 3rem;
  width: 300px;

  padding: 0.5rem;

  outline: none;
  border: none;

  background-color: white;

  color: black;
  font-size: 1rem;

  &:focus {
    border: 3px solid $blue-button-color;
  }
}

.newsletterButton {
  height: 3rem;
  width: 80px;

  // padding: 0.5rem;

  background-color: $blue-button-color;

  color: white;
  font-size: 2rem;

  border: none;

  &:hover {
    cursor: pointer;
    background-color: $blue-button-color - #222;
  }
}

@media (max-width: 550px) {
  .newsletterInput {
    width: 250px;
  }
}
