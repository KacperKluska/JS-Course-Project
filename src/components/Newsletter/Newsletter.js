import React, { useState } from "react";
import "./style.scss";

const Newsletter = () => {
  const [email, setEmail] = useState("");

  const sendEmail = (event) => {
    event.preventDefault();
    alert(`Sent email to ${email}`);
    setEmail("");
  };

  return (
    <div className="newsletter">
      <b>Sign in for our Newsletter</b>
      <label style={{ textAlign: "start" }}>
        Sign in if you want to be informed about sales and new collections.
      </label>
      <form className="newsletterForm" onSubmit={sendEmail}>
        <input
          className="newsletterInput"
          type="email"
          placeholder="Your email..."
          required
          value={email}
          onChange={(e) => {
            setEmail(e.target.value);
          }}
        ></input>
        <button type="submit" className="newsletterButton">
          Sign in
        </button>
      </form>
    </div>
  );
};

export default Newsletter;
