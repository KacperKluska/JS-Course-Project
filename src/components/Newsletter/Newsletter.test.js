import Newsletter from "./Newsletter";

let wrapper;
const setState = jest.fn();
const useStateSpy = jest.spyOn(React, "useState");
useStateSpy.mockImplementation((init) => [init, setState]);

beforeEach(() => {
  wrapper = mount(<Newsletter />);
});

afterEach(() => {
  jest.clearAllMocks();
});

test("simulate click", () => {
  // const spyFunction = jest.fn();
  console.log(wrapper.find("button"));
  wrapper
    .find("input")
    .simulate("change", { target: { value: "test@gmail.com" } });
  //   wrapper.find(".newsletterButton").simulate("click");
  //   expect(setState).toHaveBeenCalledWith("test@gmail.com");
});

test("Newsletter snapshot", () => {
  expect(wrapper).toMatchSnapshot();
});
