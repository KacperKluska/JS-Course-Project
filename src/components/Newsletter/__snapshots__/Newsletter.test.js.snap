// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter snapshot 1`] = `
<div
  className="newsletter"
>
  <b>
    <h3>
      Sign in for our Newsletter
    </h3>
  </b>
  <label
    style={
      Object {
        "textAlign": "start",
      }
    }
  >
    Sign in if you want to be informed about sales and new collections.
  </label>
  <form
    className="newsletterForm"
    onSubmit={[Function]}
  >
    <input
      className="newsletterInput"
      onChange={[Function]}
      placeholder="Your email..."
      required={true}
      type="email"
      value=""
    />
    <button
      className="newsletterButton"
      type="submit"
    >
      Sign in
    </button>
  </form>
</div>
`;
